home/tests/test_views.py

```diff
@@ -688,3 +691,115 @@ def test_ordered_form_has_all_expected_options(self, admin_client):
+class TestPageViewReportView:
+    def create_content_page(self):
+        """
+        Helper function to create pages needed for each test.
+        """
+        home_page = HomePage.objects.first()
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+
+        bodies = [WABody("default page", [WABlk("default body")])]
+
+        content_page = PageBuilder.build_cp(
+            parent=main_menu, slug="default-page", title="default page", bodies=bodies
+        )
+        return content_page
+
+    def test_template_rendering(self, admin_client):
+        """
+        The correct template is used and context data is rendered properly
+        """
+        url = reverse("page_view_report")
+        response = admin_client.get(url)
+        asserts.assertTemplateUsed(response, "reports/page_view_report.html")
+        asserts.assertContains(response, "Page views")
+
+    def test_platform_filter(self):
+        """
+        The filter returns correct data platform.
+        """
+        content_page = self.create_content_page()
+
+        PageView.objects.create(
+            timestamp="2023-01-01",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="web",
+        )
+        PageView.objects.create(
+            timestamp="2024-01-01",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="whatsapp",
+        )
+        PageView.objects.create(
+            timestamp="2024-01-15",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="whatsapp",
+        )
+
+        filterset = {"platform": "web"}
+        filter_set = PageViewFilterSet(filterset, queryset=PageView.objects.all())
+        filtered_queryset = filter_set.qs
+
+        assert filtered_queryset.count() == 1
+
+    def test_page_filter(self):
+        """
+        The filter returns correct data according to page.
+        """
+        content_page = self.create_content_page()
+
+        PageView.objects.create(
+            timestamp="2023-01-01",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="web",
+        )
+        PageView.objects.create(
+            timestamp="2024-01-01",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="whatsapp",
+        )
+        PageView.objects.create(
+            timestamp="2024-01-15",
+            page_id=content_page.id,
+            revision_id=content_page.get_latest_revision().id,
+            platform="whatsapp",
+        )
+
+        filterset_data = {"page": content_page.id}
+        filter_set = PageViewFilterSet(filterset_data, queryset=PageView.objects.all())
+        filtered_queryset = filter_set.qs
+
+        assert filtered_queryset.count() == 3
```

Comment:
I'm not sure that this test is actually testing anything. We created a single page, only put pageviews on that page, and then filtering on that single page? I think for this test, we probably want to make two pages, have pageviews on both, and then ensure that when we filter by page we only get the pageviews for the one page and not the other.